### PR TITLE
feat(scan): not-a-beer detection — filter OFF by category + dedicated UI (Issue #798)

### LINKED ISSUE #798: feat(scan): not-a-beer detection — filter OFF by category + ScanLookupNotABeerError (D)
Sub-PR 2/3 of [Epic] #794.

## Why

If a user scans a non-beer barcode (soda, water, random product), OpenFoodFacts returns a hit with non-beer category, and Brasse-Bouillon currently displays it as if it were a beer (wrong style, wrong vocabulary). Demo embarrassment risk: jury scans a Coca-Cola.

## Acceptance criteria

- [ ] Backend: \`ScanLookupService\` checks the OFF \`categories_tags\` against an allowlist (e.g. \`en:beers\`, \`en:beverages-with-fermentation\`, \`fr:bieres\`); if absent, throw a new \`ScanLookupNotABeerError\` carrying the product name + raw category
- [ ] Mobile: catch the new error, show a dedicated screen \"Vous avez scanné &lt;product name&gt; — ce n'est pas une bière\" with CTA \"Scanner une bière\" returning to the scan view
- [ ] Backend unit test: OFF returns soda → throws \`NotABeerError\`; OFF returns beer → no throw
- [ ] Mobile component test: NotABeerError → dedicated UI rendered + back-CTA wired

## Scope

- \`packages/api/src/scan/services/scan-lookup.service.ts\`
- \`packages/api/src/scan/dtos/...\` (new error type)
- \`packages/mobile-app/src/features/scan/application/scan-lookup.use-cases.ts\`
- \`packages/mobile-app/src/features/scan/presentation/BeerInfoCardScreen.tsx\`
- API + mobile tests

## Out of scope

- Multi-category support (wine, cider) — captured in #795 epic

### PR DESCRIPTION
**Pull request by benoit-bremaud**
Sub-PR 2/3 of [Epic] #794 (scan jury edge cases). PR #799 (B) was sub-PR 1/3 and is already merged.

## Summary

**Backend** (`packages/api`):
- New \`NotABeerException\` (422 UnprocessableEntityException) carrying \`errorCode='NOT_A_BEER'\` + barcode + product name
- \`ScanService.lookupByBarcode\` splits the previous combined \`!found || !isBeer\` check into two distinct branches: 404 NotFound (genuinely missing) vs 422 NotABeer (real product, non-beer category). Cache-stale fallback preserved on both paths
- 2 new spec tests + existing 23 still green

**Mobile** (\`packages/mobile-app\`):
- New \`ScanLookupNotABeerError\` typed error (barcode + productName)
- Use-case translates 422 HttpError with \`errorCode='NOT_A_BEER'\` into the typed error; other 422s pass through unchanged
- \`BeerInfoCardScreen\` extends \`ErrorVariant\` with \`not_a_beer\`, renders dedicated UI: *"Tu as scanné « Coca-Cola » — ce n'est pas une bière"* + "Scanner une bière" CTA returning to the camera
- 2 new use-case tests + 3 new screen tests

## Demo impact

If the jury scans a Coca-Cola or any non-beer barcode, the app no longer renders the wrong product as a beer. Instead, it shows a clear "ce n'est pas une bière" screen with the actual product name + a CTA back to the scan camera.

## Checklist

- [x] Happy path + sad path + edge cases covered (5 new tests across 4 files)
- [x] \`tsc --noEmit\` passes (mobile + API)
- [x] Build passes (\`npm run ci:check\` green on mobile, \`lint:check\` green on API)
- [x] All tests green (570 mobile / 384 API including 2 pre-existing skipped)
- [x] No \`any\`, no inline styles introduced

Closes #798

### CODE REVIEW

**Comment by benoit-bremaud** [comment]
@Smith06S @Thais9723 @vitalikevin — For your information: PR #800 covers the second of three scan demo edge cases (backend + frontend).

When a user scans a barcode that resolves to a non-beer product (e.g. a Coca-Cola), the app now detects this on the API side and returns a clear "this is not a beer" response with the real product name. The mobile screen displays a dedicated UI: "Tu as scanné « Coca-Cola » — ce n'est pas une bière" + a button to go back to the scan camera.

This eliminates a real demo embarrassment risk: the jury could scan a soda and our app would no longer pretend it is a beer with random style attributions.

570 mobile tests + 384 API tests all green, CI green.

File: packages/mobile-app/src/features/scan/application/scan-lookup.use-cases.ts [hunk 1]
```diff
@@ -118,7 +139,30 @@ export async function lookupBeerByBarcode(
       if (error.status === 503) {
         throw new ScanLookupServiceUnavailableError(barcode);
       }
+      if (error.status === 422 && isNotABeerDetails(error.details)) {
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/mobile-app/src/features/scan/application/scan-lookup.use-cases.ts:142]: **P1 — Implemented** (commit 8058902 on rebased branch)

You're absolutely right — the global `HttpExceptionFilter` was rewriting every `HttpException` response to the standardized `{statusCode, message, timestamp, path}` envelope, so `errorCode` / `barcode` / `productName` from `NotABeerException` were silently dropped. The mobile 422 mapping branch would never match in the real flow.

Two-step fix:

1. **`HttpExceptionFilter`** — now spreads any non-reserved fields from object exception responses alongside the canonical envelope. Reserved keys (`statusCode`, `error`, `timestamp`, `path`) keep their canonical values and cannot be spoofed by the exception payload (covered by a dedicated edge-case test).
2. **`NotABeerException`** — now also carries a human-readable `message` string so the filter has something to extract (otherwise the message would default to "Internal Server Error").

Verified:
- New `http-exception.filter.spec.ts` (6 tests): string / object / array message extraction + fallback + structured-field preservation (Issue #798 case) + reserved-key spoof protection
- All 388 API tests + 570 mobile tests green
- API lint clean (0 errors, 1 pre-existing warning unrelated)

The mobile-side mapping at line 142 (`error.details.errorCode === 'NOT_A_BEER'`) now correctly matches the production response shape.
